File: Frontend/src/types/auditoria.ts
export interface LogAccion {
  id: string;
  usuario_id: string | null;
  usuario_nombre: string | null;
  usuario_correo: string | null;
  accion: string;
  entidad: string;
  entidad_id: string | null;
  ip: string | null;
  timestamp: string;
}
export interface LogAccionPage {
  items: LogAccion[];
  total: number;
  skip: number;
  limit: number;
}

File: Backend/api/schemas/auditoria.py
from pydantic import BaseModel
from typing import Optional, List
from uuid import UUID
from datetime import datetime
class LogAccionResponse(BaseModel):
    id: UUID
    usuario_id: Optional[UUID]
    usuario_nombre: Optional[str] = None
    usuario_correo: Optional[str] = None
    accion: str
    entidad: str
    entidad_id: Optional[UUID]
    ip: Optional[str]
    timestamp: datetime
    model_config = {"from_attributes": True}
class LogAccionPage(BaseModel):
    items: List[LogAccionResponse]
    total: int
    skip: int
    limit: int
